index.js

```diff
@@ -8,6 +8,7 @@ const debug = require('debug')('jsonrpc2')
 const split = require('split2')
 const once = require('once')
 const uid = require('uid2')
+const opentracing = require('opentracing')
```

Comment:
I'm not sure if `opentracing` should be a dependency of this library.
I believe it should be passed as either:

a) a dependency (following a certain interface) that can be injected on instantiation time
b) a middleware that does the reporting

Reply:
I agree that it shouldn't be an explicit requirement here.  I was trying to think of other options, but I realized that it itself is an interface.  So if you call `globalTracer` it returns a no-op.

I looked briefly at writing a middleware but it didn't work out for some reason.  Lemme verify that again.